src/lib/translators/mongo.ts

```diff
@@ -527,6 +527,91 @@ function transformSplit(step: Readonly<S.SplitStep>): MongoStep {
   ];
 }
 
+/** transform a 'statistics' step into corresponding mongo steps */
+function transformStatistics(step: Readonly<S.StatisticsStep>): MongoStep {
+  // Some useful formula:
+  const quintilesIndexFormula: any = {
+    maxIndex: { $subtract: ['$count', 1] },
+    lastQuintileIndex: { $multiply: [{ $divide: ['$count', 10] }, 9] },
+    lastQuartileIndex: { $multiply: [{ $divide: ['$count', 4] }, 3] },
+    medianIndex: { $divide: ['$count', 2] },
+    firstQuartileIndex: { $divide: ['$count', 4] },
+    firstQuintileIndex: { $divide: ['$count', 10] },
+    minIndex: 0,
+  };
+  const varianceFormula: any = { $subtract: ['$average_sum_square', { $pow: ['$average', 2] }] };
+  const otherStatisticsFormula: any = {
+    count: 1,
+    average: 1,
+    variance: varianceFormula,
+    std: {
+      // std = square_root(N/(N-1) * variance) (With N = count) to get the unbiased version of std
+      $pow: [
+        { $multiply: [{ $divide: ['$count', { $subtract: ['$count', 1] }] }, varianceFormula] },
+        0.5,
+      ],
+    },
+  };
+  const isQuintileStatistics = (statistic: string): boolean =>
+    Object.keys(quintilesIndexFormula).includes(`${statistic}Index`);
+  // ---
+  // Some explicit conditions to compute average or columns square:
+  const computeColumnSquare = (step: Readonly<S.StatisticsStep>): boolean =>
+    step.statistics.includes('variance') || step.statistics.includes('std');
+  const computeAverage = (step: Readonly<S.StatisticsStep>): boolean =>
+    step.statistics.includes('average') ||
+    step.statistics.includes('variance') ||
+    step.statistics.includes('std');
+  // ---
+
+  const stepQuantileIndex: any = { $addFields: {} };
+  let statistic: string;
+  for (statistic of step.statistics) {
+    if (isQuintileStatistics(statistic)) {
+      stepQuantileIndex.$addFields[`${statistic}Index`] = {
+        $trunc: quintilesIndexFormula[`${statistic}Index`], //TODO: it would be more canonical to use a round version of index here. Migrating to Mongo 4.2 will allow use to use the $round step.
+      };
+    }
+  }
+  const stepProjectStatistics: any = { $project: {} };
+  for (statistic of step.statistics) {
+    if (isQuintileStatistics(statistic)) {
+      stepProjectStatistics.$project[statistic] = {
+        $arrayElemAt: ['$data', `$${statistic}Index`],
```

Comment:
+1 for null values, I add a step to filter these null value.

For the median definition, I know the one you are proposing, but I was actually using an other one where their values are actual values in the data: `p-th quantile of order n is the value of the data which is greater *or equal* of p/n% of other values and smaller (1-p/n)% of other values`. (I was using it because it was easier to implement :p )